src/main/webapp/script.js

```diff
@@ -90,17 +94,39 @@ function drawRegionsMap(censusDataArray, title) {
 
 // createDataArray takes in the data array returned by the census API
 // and reformats it into a data table for the visualization API.
-function createDataArray(censusDataArray) {
+function createDataArray(censusDataArray, description) {
   const vizDataArray = [];
-  censusDataArray.forEach((state) => {
-    vizDataArray.push([state[0], state[1]/1000]);
-  });
-  // Changes the header of the vizDataArray to match the Visualization API
+  // If the length of the census data array is four, the census call has
+  // generated a percentage and total population so we do a calculation
+  //  to find the % of the total.If the length is not four then we can
+  // just return given population.
+  if (censusDataArray[0].length === 4) {
```

Comment:
this is very brittle. Is there some other way (like test in the column header) to identify what different columns are?

Reply:
Could you elaborate on what you mean by test in the column header?

 In the displayVisualization function, we test to make sure that the query is not a county function so we can return the appropriate error response. County queries and percentage queries are the only types of queries we're dealing with that have four columns, so if the query has four columns here then it must be a percentage query. There are other ways to test for it at this point, like creating a list of known census data id's and putting them into a map or array. Is that what you mean? This would also be repeated logic from the javascript backend and would also be just as hacky given we have no formal algorithm right now.

Reply:
sorry wrong choice of words. I meant if we could determine from the header which column is which

Reply:
Yes, we could keep track of what headers correspond to percentage calls. Eg: 157E and 126E are headers that correspond percentage calls for working and moving. 

We could keep these in an array and then check to see if any of the headers appear in the array. It would be more time consuming and still hacky because we're hard coding the values but more maintainable to build on.

Reply:
If you move that into separate mapping, it can be hard coded now but later maybe we can find a way to get that information directly from the census data api. Also it is not depending on number of columns or order of columns which makes it a little more robust.

Reply:
Fixed!